### DISCUSSION TITLE
How to modify topbar links ?

### DISCUSSION THREAD

**Discussion by Simeon-Ferrara** [chamilo/chamilo-lms#5181] [General] [answered]
Hello,

As an experienced developer who is new to Chamilo, I am trying to modify the "My courses" link in the topbar to redirect it to the `courses.php` page, instead of the default `user_portal.php`.

<img width="444" alt="Capture d’écran 2024-02-22 à 11 43 01" src="https://github.com/chamilo/chamilo-lms/assets/129834935/b8304201-fcfe-441a-91a5-b86ae061e9d8">

I've created a new template file at `/main/template/overrides/layout/topbar.tpl` and updated the corresponding link as follows:
```html
 <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
            <ul class="nav navbar-nav">
                <li class="active"><a href="{{ _p.web }}courses.php"> {{ "MyCourses"|get_lang }}</a></li>
...               
```

After making these changes, I cleared my browser and the Chamilo cache following the method described in the Chamilo documentation: https://docs.chamilo.org/developer-guide/theming_through_templates/cleaning_the_cache.
Despite this, the link remains unchanged.

I've checked the cached files and can confirm that my changes are saved, and the topbar should display the correct link, but it does not reflect the changes.
<img width="497" alt="Capture d’écran 2024-02-22 à 11 51 05" src="https://github.com/chamilo/chamilo-lms/assets/129834935/5a7e9d90-94c5-4d46-8933-1ac287b3fcbe">
<img width="483" alt="Capture d’écran 2024-02-22 à 12 04 37" src="https://github.com/chamilo/chamilo-lms/assets/129834935/ed00a5c0-3948-48b1-ba63-21edc6e9c2d3">

Is there something I'm missing, or an additional step I need to take for the changes to take effect?
Also I am curious to know if there's a way to accomplish this without altering the code.

**Comment by etree2024** [comment]
Are you running OPCACHE on the server? if yes get the OPCACHE Gui from git hub.

**Comment by ywarnier** [comment, marked as answer]
Hi @Simeon-Ferrara 

1st, there's no way to accomplish this without altering the code (so you know).
2nd, your procedure seems OK, but I think the topbar.tpl file is for the special admin bar that you can enable in the administration settings (search for  `Show admin toolbar` in the settings)
3rd I believe what you're looking for (and it's worst because it is not themed) is main/inc/lib/banner.lib.php on lines 46 and 49 (in the latest code, but you can just search for "user_portal.php"). That should work without cache refresh.

**Reply by Simeon-Ferrara** [reply]
Hi!
I was able to achieve what I wanted, but I forgot to mark the thread as resolved.

Thanks for your answer anyway!
Hope it can help someone else ✌️
